### PR TITLE
[WIP] Update some packages to newer or latest versions

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Comment by iarspider** [comment]
please test

things will probably break :)

File: libxslt.spec [hunk 1]
```diff
@@ -1,11 +1,8 @@
-### RPM external libxslt 1.1.28
+### RPM external libxslt 1.1.33
 
-%define tag 195f644cf7b5e93208b033fa7c5501c41cf82f59
-%define branch master
-%define github_user cms-externals
-Source0: git+https://github.com/%github_user/%{n}.git?obj=%{branch}/%{tag}&export=%{n}-%{realversion}&output=/%{n}-%{realversion}.tgz
+Source0: http://xmlsoft.org/sources/%{n}-%{realversion}.tar.gz
```
Comment (smuzaffar) [inline libxslt.spec]: why not https://github.com/GNOME/libxslt/tree/v1.1.35 ?

Comment (iarspider) [reply libxslt.spec]: No particular reason. Done.

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
@iarspider bad idea to update gcc for production arch at this point. why it is needed?

**Comment by iarspider** [comment]
@smuzaffar just syncing with spack. I will revert that change.

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
The issue with libxml/libxslt is weird: after fixing include path I see that configure script is only adding `-L<path-to-libxml2>`, but not `-lxml2` to the makefiles. I will try to switch both libs to CMake, maybe this will help.

**Comment by smuzaffar** [comment]
please test

**Comment by iarspider** [comment]
@smuzaffar I will run the bot test once the local build works (it does not at the moment)

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
Todo: ~~py3-astroid 2.11.6, py3-lazy-object-proxy 1.7.1, py3-pylint 2.14.4 (NB: check if py3-tensorflow is fine with  py3-typing-extensions version that pylint requires)~~ py3-luigi 3.1.0

**Comment by iarspider** [comment]
I need to redo the PR since we are stuck with old TF. Closing it for now.

**Pull request closed without merging**